ton_client/client/src/tvm/run_message.rs

```diff
@@ -111,25 +188,14 @@ pub async fn run_executor(
     context: std::sync::Arc<ClientContext>,
     params: ParamsOfRunExecutor,
 ) -> ClientResult<ResultOfRunExecutor> {
-    let account = params
-        .account
-        .map(|string| deserialize_cell_from_base64(&string, "account"))
-        .transpose()?
-        .map(|tuple| tuple.1);
     let message = deserialize_object_from_base64::<Message>(&params.message, "message")?.object;
+    let msg_address = message.dst().ok_or(Error::invalid_message_type())?;
+    let (account, original_balance) = params
+        .account
+        .get_account(&context, msg_address.clone())?;
     let options = ResolvedExecutionOptions::from_options(&context, params.execution_options)?;
 
-    // if no account provided use AccountNone
-    let account = match account {
-        Some(account) => account,
-        None => ton_block::Account::AccountNone
-            .write_to_new_cell()
-            .unwrap()
-            .into(),
-    };
-
     let account_copy = account.clone();
-    let msg_address = message.dst().ok_or(Error::invalid_message_type())?;
     let contract_info = move || async move {
```

Comment:
I don't understand why account is passed using callback. Is there are some reasons for it?

Reply:
Account data is needed only in case of error. So callback is needed to keep positive branch optimal without parsing account data from cell. It is more clear in `processing` module where we have to download account